server/controllers/postUser.js

```diff
@@ -1,43 +1,21 @@
 const postUserQuery = require('../database/queries/addUsers');
-
-const joi = require('@hapi/joi');
-const validationObject = {
-  username: joi // username should be string and at least it contains 2 characters
-    .string()
-    .min(2)
-    .max(40),
-  email: joi // email should be string and valid and its tld should be .com
-    .string()
-    .email({
-      minDomainSegments: 2,
-      tlds: { allow: ['com'] },
-    })
-    .min(10),
-  password: joi // password must be string, at least have 6 characters contains capital letters, small letter and numbers
-    .string()
-    .alphanum()
-    .min(6)
-    .pattern(new RegExp('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])')),
-};
-
-const schema = joi.object(validationObject);
+const checkValidation = require('../helpers/validation');
 
 exports.postUser = (req, res) => {
-  const { error, value } = schema.validate(req.body);
-  const { username, email, password } = value;
+  //validate user information
+  const { error } = checkValidation(req.body);
   if (error) {
-    res.status(400).json({ error: error.details[0].message });
-  } else {
-    try {
-      const userInfo = {
-        username,
-        email,
-        password,
-      };
-      postUserQuery(userInfo);
-      res.status(200).json({ message: 'user  created succefully' });
-    } catch (err) {
-      res.status(400).json({ message: 'There is no valid information' });
-    }
+    return res.status(400).json({ message: error.details[0].message });
+  }
+  // store user information to datbase
+  try {
+    postUserQuery(req.body, (postError) => {
+      if (postError) {
+        return res.status(400).json({ message: postError.message });
+      }
+      return res.status(200).json({ message: 'user  created successfully' });
+    });
+  } catch (err) {
+    res.status(400).json({ message: 'There is no valid information' });
```

Comment:
you can use `async await` instead of callback for for the `postUserQuery` hence you already using try catch....

Reply:
@ghassanmas I've changed it, could you check please if I did it in the right way?